pkg/storage/tsdb/block/block_test.go

```diff
@@ -414,6 +414,56 @@ func TestMarkForNoCompact(t *testing.T) {
 	}
 }
 
+func TestUnMarkForNoCompact(t *testing.T) {
+	testutil.VerifyNoLeak(t)
+	ctx := context.Background()
+	tmpDir := t.TempDir()
+	for tname, tcase := range map[string]struct {
+		setupTest     func(t testing.TB, id ulid.ULID, bkt objstore.Bucket)
+		expectedError func(id ulid.ULID) error
+	}{
+		"unmark existing block should succeed": {
```

Comment:
Should were add a test case for unmarking a block which doesn't exist?

Reply:
The second test "unmark non-existing block should fail", is this what you want?

Reply:
crap, this was written in autopilot. I meant write a test which tries to unmark a block which doesn't have a marker

Reply:
👌 will add it in the next PR